parsedatetime/__init__.py

```diff
@@ -863,7 +898,166 @@ def _quantityToReal(self, quantity):
+                return numbers[0] * numbers[1] + numbers[2]
+        elif len(numbers) == 2:
+            if numbers[1] % 100 == 0:
+                return numbers[0] * numbers[1]
+            else:
+                return numbers[0] + numbers[1]
+        else:
+            return numbers[0]
+
+    def word_to_num(self, number_sentence):
+        if not number_sentence:
+            return 1.0
+        try:
+            return float(number_sentence.replace(',', '.'))
+        except ValueError:
+            pass
+
+        number_sentence = number_sentence.replace('-', ' ')
+        number_sentence = number_sentence.lower()  # converting input to lowercase
+        if 'ignore' in self.ptc.numbers:
+            number_sentence = number_sentence.replace(' '+self.ptc.numbers['ignore']+ ' ', ' ')
+
+        if (number_sentence.isdigit()):  # return the number if user enters a number string
+            return float(number_sentence)
+
+        clean_numbers = []
+        clean_decimal_numbers = []
+
+        # removing and, & etc.
+
+        str = number_sentence.strip() + ' '
+        m = self.ptc.CRE_MONTH.search(str)
+        if m is not None:
+            str = str.replace(m.group(), ' ')
+        m = self.ptc.CRE_WEEKDAY.search(str)
+        if m is not None:
+            str = str.replace(m.group(), ' ')
+        str = ' '+ str + ' '
+        while 1:
+            m = self.ptc.CRE_NUM.search(str)
+            if m is not None:
+                clean_numbers.append(m.group())
+                str = str[m.end():].strip()
+            else:
+                break
+
+        # Error message if the user enters invalid input!
+        if len(clean_numbers) == 0:
+            raise ValueError("No valid number words found! Please enter a valid number word (eg. two million twenty three thousand and forty nine)")
+
+        # Error if user enters million,billion, thousand or decimal point twice
+        if clean_numbers.count('thousand') > 1 or clean_numbers.count('million') > 1 or clean_numbers.count(
+            'billion') > 1 or clean_numbers.count('point') > 1:
+            raise ValueError("Redundant number word! Please enter a valid number word (eg. two million twenty three thousand and forty nine)")
+
+        # separate decimal part of number (if exists)
+        if clean_numbers.count('point') == 1:
+            clean_decimal_numbers = clean_numbers[clean_numbers.index('point') + 1:]
+            clean_numbers = clean_numbers[:clean_numbers.index('point')]
+
+        billion_index = clean_numbers.index('billion') if 'billion' in clean_numbers else -1
+        million_index = clean_numbers.index('million') if 'million' in clean_numbers else -1
+        thousand_index = clean_numbers.index('thousand') if 'thousand' in clean_numbers else -1
+
+        if (thousand_index > -1 and (thousand_index < million_index or thousand_index < billion_index)) or (
+            million_index > -1 and million_index < billion_index):
+            raise ValueError(
```

Comment:
In our library, we should not throw exceptions when there is malformed number. When number is ill formed, just return date or range not found or invalid.

Reply:
I worked with three digit numbers, no change was made to the higher value numbers